### PR TITLE
Add API Endpoints

### PR DESCRIPTION
**Pull request by laiifuu**
## Hi :wave: 
For this PR, we: 
- Created an API endpoint to list all posts for a user.
- Created an API endpoint to list all comments for a user's post.
- Created an API endpoint to add a comment to a post.

### CODE REVIEW

File: app/controllers/api_controller.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    @comments = Comment.where(post_id: params[:post_id])
+    render json: @comments
+  end
+
+  def create_comment
+    text = params[:text]
+    post_id = params[:post_id]
+    comment = Comment.create(text:, author: current_user, post_id:)
+    render json: comment, status: 200
+  end
```
Comment (youmari) [inline app/controllers/api_controller.rb]: - [ ] when I tried to create a comment using the api and I was signed in  it returns the following error so please try to make sure to fix that 
Screenshot 2023-02-03 at 16 37 41

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @laiifuu @tarikbouari , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Great commit messages
✔️ PRO PR
✔️ No linter error 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/api_controller.rb (hunk 1 above)
Comment (laiifuu) [reply app/controllers/api_controller.rb]: @youmari Hi, could you tell me how you tried to add the comment?

Comment (youmari) [reply app/controllers/api_controller.rb]: @laiifuu Salam 3alaykoum When I added from the browser like that `http://127.0.0.1:3000/users/1/api/v1/posts/1/comments/toto`it worked but if I use a app for apis like postman or thunder it didn't work and the api should work anywhere

Comment (laiifuu) [reply app/controllers/api_controller.rb]: Okay, thanks

Comment (youmari) [reply app/controllers/api_controller.rb]: @laiifuu welcome and it should be `post` method not a `get`

**Review by afanimartin — APPROVED** [review]
Hi @laiifuu, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🎉 🎉 🎉 
- changes made as requested 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

### Martin Afani
#### Slack: whiz25

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**